src/Search/Extensions/IndexableExtension.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace SilverStripe\FullTextSearch\Search\Extensions;
+
+use SilverStripe\ORM\DataExtension;
+
+/**
+ * Checks if a DataObject is publically viewable thus able to be added or retrieved from a publically searchable index
+ */
+class IndexableExtension extends DataExtension
+{
+    const INDEXABLE_STATE_UNCHECKED = 'INDEXABLE_STATE_UNCHECKED';
+
+    const INDEXABLE_STATE_ALLOWED = 'INDEXABLE_STATE_ALLOWED';
+
+    const INDEXABLE_STATE_DISALLOWED = 'INDEXABLE_STATE_DISALLOWED';
+
+    /**
+     * @var string
+     */
+    protected $indexableState;
```

Comment:
I'm trying to comprehend why this is a safe place to store this state - are `DataExtension` objects persisted against `DataObjects` across multiple queries / `DataLists` somehow? If not, this runs the risk of being recomputed in cases where it's called multiple times at different layers. Not sure if that statement holds up in real-world cases, just a little concerned.

Reply:
Ingo recommended changing this from a DataExtension to a singleton instead to hold all the cached results, so will do that